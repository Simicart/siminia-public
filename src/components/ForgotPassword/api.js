export const resetPasswordRequest = () => {
    // TODO: Handle reset password.
    return Promise.resolve();
};
