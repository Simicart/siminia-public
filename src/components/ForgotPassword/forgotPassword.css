.instructions {
    padding: 0.5rem 1.5rem 0;
    margin-bottom: 1.5rem;
    font-size: 0.875rem;
    font-weight: 300;
    line-height: 1.25rem;
}
