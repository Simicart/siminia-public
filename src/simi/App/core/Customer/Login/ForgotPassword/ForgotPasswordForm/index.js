import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'informed';
import Field from '@magento/venia-ui/lib/components/Field';
import TextInput from '@magento/venia-ui/lib/components/TextInput';
import { isRequired } from 'src/util/formValidators';
import { configColor } from 'src/simi/Config';
import Identify from 'src/simi/Helper/Identify';

require('./forgotPasswordForm.scss');

const ForgotPasswordForm = props => {
    const { isResettingPassword, onSubmit } = props;

    return (
        <Form className='root forgot-pass-form' onSubmit={onSubmit}>
            <Field label={Identify.__("Email Address")} required={false} after={'*'} >
                <TextInput
                    autoComplete="email"
                    field="email"
                    validate={isRequired}
                    validateOnBlur
                />
            </Field>
            <div className='buttonContainer'>
                <button priority="high" className='submitButton' type="submit" disabled={isResettingPassword}
                    style={{ backgroundColor: configColor.button_background, color: configColor.button_text_color }}>
                    {Identify.__('Submit')}
                </button>
            </div>
        </Form>
    )
}

ForgotPasswordForm.propTypes = {
    onSubmit: PropTypes.func.isRequired
}

export default ForgotPasswordForm;
