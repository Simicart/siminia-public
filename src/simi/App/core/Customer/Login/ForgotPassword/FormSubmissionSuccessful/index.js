import React from 'react';
import Identify from 'src/simi/Helper/Identify';
import { configColor } from 'src/simi/Config';

require('./formSubmissionSuccessful.scss');

const FormSubmissionSuccessful = props => {
    const { history, email } = props;
    const textMessage = Identify.__('If there is an account associated with %s you will receive an email with a link to change your password.').replace('%s', email);

    return (
        <div className="form-successful">
            <p className='text'>{textMessage}</p>
            <div className='buttonContainer'>
                <button
                    onClick={() => history.push('/')}
                    className='submitButton'
                    style={{ backgroundColor: configColor.button_background, color: configColor.button_text_color }}>
                    {Identify.__('Continue Shopping')}
                </button>
            </div>
        </div>
    );
}

export default FormSubmissionSuccessful;
