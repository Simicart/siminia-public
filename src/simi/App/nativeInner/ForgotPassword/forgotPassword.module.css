.root {
    display: grid;
    gap: 1.5rem;
    justify-items: stretch;
    padding: 1.5rem;
}

.title {
    padding-top: 0.5rem;
    text-transform: capitalize;
}

.instructions {
    font-weight: 300;
    line-height: 1.25rem;
    
}
@media (max-width:450px) {
    .root {
        padding: 1rem 10px;
    }
    .instructions {
        text-align: left;
    }
}
