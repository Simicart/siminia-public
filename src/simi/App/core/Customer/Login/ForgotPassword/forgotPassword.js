import React from 'react';
import { func, shape, string } from 'prop-types';
import Identify from 'src/simi/Helper/Identify';
import FormErrors from '@magento/venia-ui/lib/components/FormError';
import { useForgotPassword } from 'src/simi/talons/ForgotPassword/useForgotPassword';
import forgotPasswordOperations from './forgotPassword.gql';
import FormSubmissionSuccessful from './FormSubmissionSuccessful';
import ForgotPasswordForm from './ForgotPasswordForm';

require('./forgotPassword.scss');

const ForgotPassword = props => {
    const { initialValues, onCancel, history } = props;

    const talonProps = useForgotPassword({
        onCancel,
        ...forgotPasswordOperations
    });

    const {
        forgotPasswordEmail,
        formErrors,
        handleCancel,
        handleFormSubmit,
        hasCompleted,
        isResettingPassword
    } = talonProps;

    const children = hasCompleted ? (
        <FormSubmissionSuccessful email={forgotPasswordEmail} history={history} />
    ) : (
            <React.Fragment>
                <p className='instructions'>
                    {Identify.__('Enter your email below to receive a password reset link')}
                </p>
                <ForgotPasswordForm
                    initialValues={initialValues}
                    isResettingPassword={isResettingPassword}
                    onSubmit={handleFormSubmit}
                    onCancel={handleCancel} />
                <div style={{ marginTop: '15px' }}>
                    <FormErrors errors={formErrors} />
                </div>
            </React.Fragment>
        );

    return <div className='forgot-password-root'>{children}</div>;
};

export default ForgotPassword;

ForgotPassword.propTypes = {
    classes: shape({
        instructions: string,
        root: string
    }),
    initialValues: shape({
        email: string
    }),
    onCancel: func
};

ForgotPassword.defaultProps = {
    onCancel: () => { }
};
