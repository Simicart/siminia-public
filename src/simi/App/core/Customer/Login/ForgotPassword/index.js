import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import ForgotPasswordForm from './ForgotPasswordForm';
import FormSubmissionSuccessful from './FormSubmissionSuccessful';
import TitleHelper from 'src/simi/Helper/TitleHelper'
import Identify from 'src/simi/Helper/Identify'
import {forgotPassword} from 'src/simi/Model/Customer'
import {showFogLoading, hideFogLoading} from 'src/simi/BaseComponents/Loading/GlobalLoading'
import {showToastMessage} from 'src/simi/Helper/Message';

require('./forgotPassword.scss');

class ForgotPassword extends Component {

    state  = {resetSubmited: false}

    static propTypes = {
        classes: PropTypes.shape({
            instructions: PropTypes.string
        }),
        onClose: PropTypes.func.isRequired
    };

    handleFormSubmit = ({ email }) => {
        showFogLoading()
        forgotPassword(this.resetSubmited, email)
    }

    resetSubmited = (data) => {
        hideFogLoading()
        if (data && !data.errors) {
            let text = '';
            if (data.message) {
                const messages = data.message;
                for (const i in messages) {
                    const message = messages[i];
                    text += message + ' ';
                }
            }
            this.successMessage = text
            this.setState({ resetSubmited: true})
        } else {
            let messages = ''
            data.errors.map(value => {
                messages +=  value.message
            })
            showToastMessage(messages)
            this.setState({ resetSubmited: false})
        }
    }

    handleContinue = () => {
        this.setState({resetSubmited: false})
        if (this.props.onClose)
            this.props.onClose();
    };

    render() {
        const { email } = this.props;
        const {resetSubmited} = this.state

        if (resetSubmited) {
            return (
                <FormSubmissionSuccessful
                    email={email}
                    onContinue={this.handleContinue}
                    successMessage={this.successMessage}
                />
            );
        }

        return (
            <Fragment>
                {TitleHelper.renderMetaHeader({
                    title:Identify.__('Forgot password')
                })}
                <p className='instructions'>
                    {Identify.__('Enter your email below to receive a password reset link')}
                </p>
                <ForgotPasswordForm
                    onSubmit={this.handleFormSubmit}
                />
            </Fragment>
        );
    }
}

export default ForgotPassword;
