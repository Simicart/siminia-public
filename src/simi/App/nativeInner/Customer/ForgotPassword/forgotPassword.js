import React, { Fragment } from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { func, shape, string } from 'prop-types';

import { useForgotPassword } from '../../../../talons/ForgotPassword/useForgotPassword';

import FormErrors from '@magento/venia-ui/lib/components/FormError';
import { useStyle } from '@magento/venia-ui/lib/classify';
import ForgotPasswordForm from './ForgotPasswordForm';
import FormSubmissionSuccessful from './FormSubmissionSuccessful';

import forgotPasswordOperations from './forgotPassword.gql';

import defaultClasses from './forgotPassword.module.css';

const ForgotPassword = props => {
    const { initialValues, onCancel } = props;

    const { formatMessage } = useIntl();
    const talonProps = useForgotPassword({
        onCancel,
        ...forgotPasswordOperations
    });

    const {
        forgotPasswordEmail,
        formErrors,
        handleCancel,
        handleFormSubmit,
        hasCompleted,
        isResettingPassword,
        recaptchaWidgetProps
    } = talonProps;

    const classes = useStyle(defaultClasses, props.classes);
    const INSTRUCTIONS = formatMessage({
        id: 'forgotPassword.instructions',
        defaultMessage:
            'Please enter the email address associated with this account.'
    });
    const children = hasCompleted ? (
        <FormSubmissionSuccessful email={forgotPasswordEmail} />
    ) : (
        <Fragment>
            <h2 data-cy="ForgotPassword-title" className={classes.title}>
                <FormattedMessage
                    id={'RECOVER PASSWORD'}
                    defaultMessage={'Recover Password'}
                />
            </h2>
            <p
                data-cy="ForgotPassword-instructions"
                className={classes.instructions}
            >
                {INSTRUCTIONS}
            </p>
            <ForgotPasswordForm
                initialValues={initialValues}
                isResettingPassword={isResettingPassword}
                onSubmit={handleFormSubmit}
                onCancel={handleCancel}
                recaptchaWidgetProps={recaptchaWidgetProps}
            />
            <FormErrors errors={formErrors} />
        </Fragment>
    );

    return <div className={classes.root}>{children}</div>;
};

export default ForgotPassword;

ForgotPassword.propTypes = {
    classes: shape({
        instructions: string,
        root: string
    }),
    initialValues: shape({
        email: string
    }),
    onCancel: func
};

ForgotPassword.defaultProps = {
    onCancel: () => {}
};
