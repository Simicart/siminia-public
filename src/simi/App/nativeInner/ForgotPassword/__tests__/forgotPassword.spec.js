import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import { useForgotPassword } from '@magento/peregrine/lib/talons/ForgotPassword/useForgotPassword';

import ForgotPassword from '../forgotPassword';

jest.mock('../FormSubmissionSuccessful', () => props => (
    <mock-FormSubmissionSuccessful {...props} />
));
jest.mock('../ForgotPasswordForm', () => props => (
    <mock-ForgotPasswordForm {...props} />
));

jest.mock(
    '@magento/peregrine/lib/talons/ForgotPassword/useForgotPassword',
    () => ({
        useForgotPassword: jest.fn().mockReturnValue({
            forgotPasswordEmail: 'gooseton@goosemail.com',
            formErrors: [],
            handleCancel: jest.fn(),
            handleFormSubmit: jest.fn(),
            hasCompleted: false,
            isResettingPassword: false
        })
    })
);

test('should render properly', () => {
    const tree = createTestInstance(
        <ForgotPassword
            initialValues={{ email: 'gooseton@goosemail.com' }}
            onCancel={jest.fn()}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('should render successful view if hasCompleted is true', () => {
    useForgotPassword.mockReturnValueOnce({
        forgotPasswordEmail: 'gooseton@goosemail.com',
        formErrors: [],
        handleCancel: jest.fn(),
        handleFormSubmit: jest.fn(),
        hasCompleted: true,
        isResettingPassword: false
    });

    const tree = createTestInstance(
        <ForgotPassword
            initialValues={{ email: 'gooseton@goosemail.com' }}
            onCancel={jest.fn()}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
