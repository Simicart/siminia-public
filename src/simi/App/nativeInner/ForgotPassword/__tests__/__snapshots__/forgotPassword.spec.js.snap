// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<div>
  <h2>
    <mock-FormattedMessage
      defaultMessage="Recover Password"
      id="forgotPassword.recoverPasswordText"
    />
  </h2>
  <p>
    Please enter the email address associated with this account.
  </p>
  <mock-ForgotPasswordForm
    initialValues={
      Object {
        "email": "gooseton@goosemail.com",
      }
    }
    isResettingPassword={false}
    onCancel={[MockFunction]}
    onSubmit={[MockFunction]}
  />
</div>
`;

exports[`should render successful view if hasCompleted is true 1`] = `
<div>
  <mock-FormSubmissionSuccessful
    email="gooseton@goosemail.com"
  />
</div>
`;
